spinalcordtoolbox/aggregate_slicewise.py

```diff
@@ -293,6 +293,8 @@ def aggregate_per_slice_or_level(metric, mask=None, slices=[], levels=[], distan
             slicegroups = [tuple([val for sublist in slicegroups for val in sublist])]  # flatten into single tuple
             # vertgroups = [(2, 3, 4)]
             vertgroups = [tuple([level for level in levels])]
+        # Intersection between specified slices and each element of slicegroups
+        slicegroups = [tuple(set(slicegroup) & set(slices)) for slicegroup in slicegroups]
     # aggregation based on slices
```

Comment:
I think this is relevant specifically for `perslice`, because if `slicegroups` gets reduced, then right now, `vertgroups` will be the wrong length, since it didn't also get reduced?

I added a `perslice` section to your test:

```python
    agg_metric_perslice = aggregate_slicewise.aggregate_per_slice_or_level(dummy_metrics['with float'], slices=[1, 2, 3, 4, 6],
                                                                           levels=[2, 3], perlevel=False, perslice=True,
                                                                           vert_level=dummy_vert_level,
                                                                           group_funcs=(('WA', aggregate_slicewise.func_wa),))
    assert ((1,), (2,), (3,)) == tuple(agg_metric_perslice.keys())
    assert agg_metric_perslice[(1,)] == {'VertLevel': (2,), 'DistancePMJ': None, 'WA()': 31.0}
    assert agg_metric_perslice[(2,)] == {'VertLevel': (3,), 'DistancePMJ': None, 'WA()': 39.0}
    assert agg_metric_perslice[(3,)] == {'VertLevel': (3,), 'DistancePMJ': None, 'WA()': 41.0}
```

However, with the PR as-is, this fails, because `vertgroups` never gets reduced. So, there's an empty entry for `slice=0`:

![image](https://user-images.githubusercontent.com/16181459/176472670-dcae0c5f-268d-4c6b-aaf0-ee7dd6a79074.png)

If we make the above change, then the test passes:

![image](https://user-images.githubusercontent.com/16181459/176472832-c14c4334-3500-4233-a568-fb4115d9e7ed.png)

Reply:
I can push these changes shortly. :)

Reply:
that's smart! yes please, push the changes and then we can merge so i can let the user know it is ready to use (user is a student at UofT and she has a presentation to do in a few days based on these results so she is pressed with time)

Reply:
Ahh, I see! Thank you for the clarification. :)